flexmeasures/data/models/reporting/pandas_reporter.py

```diff
@@ -0,0 +1,114 @@
+
+    def _compute(self) -> pd.Series:
+        """"""
+        # apply pandas transformations to the dataframes in `self.data``
+        self.apply_transformations()
+
+        final_output = self.data[self.final_df_output]
+        return final_output
+
+    def get_object_or_literal(self, value, method):
+        """This method allows using the dataframes as inputs of the Pandas methods that
+        are run in the transformations. Make sure that they have been created before accessed.
+
+        This works by putting the symbol `@` in front of the name of the dataframe that we want to reference.
+        For instance, to reference the dataframe test_df, which lives in self.data, we would do `@test_df`.
+
+        This functionality is disabled for methods `eval`and `query` to avoid interfering their internal behaviour
+        given that they also use `@` to allow using local variables.
+
+        Examples
+        >> self.get_object_or_literal(["@sensor_1", "@sensor_2"], "sum")
+        [[                    ...n: 0:00:00,                     ...n: 0:00:00]]
+        """
+
+        if method in ["eval", "query"]:
+            return value
+
+        if isinstance(value, str) and value.startswith("@"):
+            value = value.replace("@", "")
+            return self.data[value]
+
+        if isinstance(value, list):
+            return [self.get_object_or_literal(v, method) for v in value]
+
+        return value
+
+    def process_pandas_args(self, args, method):
+        """This method applies the function get_object_or_literal to all the arguments
+        to detect where to replace a string "@<object-name>" with the actual object stored in `self.data["<object-name>"]`.
+        """
+        for i in range(len(args)):
+            args[i] = self.get_object_or_literal(args[i], method)
+        return args
+
+    def process_pandas_kwargs(self, kwargs, method):
+        """This method applies the function get_object_or_literal to all the keyword arguments
+        to detect where to replace a string "@<object-name>" with the actual object stored in `self.data["<object-name>"]`.
+        """
+        for k, v in kwargs.items():
+            kwargs[k] = self.get_object_or_literal(v, method)
+        return kwargs
+
+    def apply_transformations(self) -> pd.Series:
+        """Convert the series using the given list of transformation specs, which is called in the order given.
+
+        Each transformation specs should include a 'method' key specifying a method name of a Pandas DataFrame.
+
+        Optionally, 'args' and 'kwargs' keys can be specified to pass on arguments or keyword arguments to the given method.
+
+        All data exchange is made through the dictionary `self.data`. The superclass Reporter already fetches and saves BeliefDataFrames
+        of the input sensors in the fields `sensor_<sensor_id>`. In case you need to perform complex operations on dataframes, you can
```

Comment:
Small nuance. I suggest rephrasing to avoid suggesting data is both fetched and saved from/to the database:

[...] fetches BeliefsDataFrames of the sensors and saves them in the `self.data` dictionary fields [...]